Security issue with tracking logged in users with websocket

It seems that any authenticated user can listen to the `topic/tracker` and thereby see the user tracking information.

See line 17 websocket security config in [WebsocketSecurityConfiguration.java](https://github.com/jhipster/generator-jhipster/blob/master/app/templates/src/main/java/package/config/_WebsocketSecurityConfiguration.java#L17)

We should configure the tracker topic to only be available for ROLE_ADMIN users with:
```
.simpDestMatchers("/topic/tracker").hasAuthority(AuthoritiesConstants.ADMIN)
```

It seems to work with authentication type `session` and `oauth2`. So I will modify my PR to include the authentication config for both  `session` and `oauth2`. And

With `xauth` I get the following authentication error in the console
```
<<< ERROR
message:Failed to send message to ExecutorSubscribableChannel[clientInboundChannel]; nested exception is org.springframework.security.access.AccessDeniedException\c Access is denied
content-length:0
```

and from the spring logs
```
[ERROR] org.springframework.web.socket.messaging.StompSubProtocolHandler - Failed to send client message to application via MessageChannel in session myaqtjdp. Sending STOMP ERROR to client.
org.springframework.messaging.MessageDeliveryException: Failed to send message to ExecutorSubscribableChannel[clientInboundChannel]; nested exception is org.springframework.security.access.AccessDeniedException: Access is denied
```

so I'll leave xauth as it is.

--- a/app/templates/src/main/java/package/config/_WebsocketSecurityConfiguration.java
+++ b/app/templates/src/main/java/package/config/_WebsocketSecurityConfiguration.java
@@ -1,5 +1,8 @@
 package <%=packageName%>.config;
 
+<%_ if (authenticationType == 'session' || authenticationType == 'oauth2') { _%>
+import <%=packageName%>.security.AuthoritiesConstants;
+<%_ } _%>
 import org.springframework.context.annotation.Configuration;
 import org.springframework.messaging.simp.SimpMessageType;
 import org.springframework.security.config.annotation.web.messaging.MessageSecurityMetadataSourceRegistry;
@@ -14,6 +17,9 @@ public class WebsocketSecurityConfiguration extends AbstractSecurityWebSocketMes
             // message types other than MESSAGE and SUBSCRIBE
             .nullDestMatcher().authenticated()
             // matches any destination that starts with /rooms/
+            <%_ if (authenticationType == 'session' || authenticationType == 'oauth2') { // ignoring authentiation check for authenticationType == 'xauth' since it does not work, see issue #2129 _%>
+            .simpDestMatchers("/topic/tracker").hasAuthority(AuthoritiesConstants.ADMIN)
+            <%_ } _%>
             .simpDestMatchers("/topic/**").authenticated()
             // (i.e. cannot send messages directly to /topic/, /queue/)
             // (i.e. cannot subscribe to /topic/messages/* to get messages sent to

